python/pyspark/ml/deepspeed/tests/test_deepspeed_distributor.py

```diff
@@ -163,6 +169,84 @@ def test_create_torchrun_command_distributed(self) -> None:
             )
             self.assertEqual(distributed_cmd_args_expected, distributed_command_with_args)
 
+def _create_basic_function():
+   def pythagoras(leg1: float, leg2: float):
+       import deepspeed
+       return (leg1 * leg1 + leg2 * leg2)**0.5
+   return pythagoras 
+
+
+class DeepspeedTorchDistributorDistributedEndToEnd(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        (cls.gpu_discovery_script_file_name, cls.mnist_dir_path) = set_up_test_dirs()
+        conf = SparkConf(loadDefaults=False)
+        for k, v in get_distributed_mode_conf().items():
+            conf = conf.set(k, v)
+        conf = conf.set(
+            "spark.worker.resource.gpu.discoveryScript", cls.gpu_discovery_script_file_name
+        )
+        sc = SparkContext("local-cluster[2,2,512]",cls.__name__,conf=conf)
+        cls.spark = SparkSession(sc)
+
+    def test_simple_function_e2e(self):
+       train_fn = _create_basic_function()
+       # Arguments for the pythagoras function train_fn
+       x = 3
+       y = 4
+       dist = DeepspeedTorchDistributor(num_gpus=2, use_gpu=False, local_mode=False)
+       output = dist.run(train_fn, x, y)
+       self.assertEqual(output, 5)
+
+    def test_pytorch_file_e2e(self):
+        path_to_train_file = "python/test_support/test_deepspeed_training_file.py"
```

Comment:
This will be an issue if someone runs the test from a different directory. Instead of committing python/test_support/test_deepspeed_training_file.py, let's just write out its contents to the /tmp file in the test setup or test case